### ISSUE TITLE
[V3 RPC] Tracking for the addition of new RPC methods

### ISSUE THREAD

**Issue by NeuroAssassin** [Cog-Creators/Red-DiscordBot#2630]
# Feature request

#### Select the type of feature you are requesting:

- [ ] Cog
- [ ] Command
- [X] API functionality

### Description of the changes
This issue will be used for the tracking of the PRs I create for adding RPC methods to Red.  Note that I have only just started this, I am only making this for core developers to keep track of my changes.  Each PR I make will be at first drafts, and then once they are finished, I will mark it as ready for review.

#### Downloader:
I will be waiting on creating RPC methods for Downloader to see the outcome of #2571.
#### Audio:
I need to sit down with audio and examine the code, but functions such as setting the jar to be internal or external, should be made, however I will discuss with Aikaterna beforehand.
#### Cleanup:
This cog probably won't receive any RPC handlers, as it deals with messages in channels, and seems out of scope with RPC.
#### Mod/Move to Core:
I have a little trouble thinking about incorporating banning, kicking and other administrative actions into RPC, as they seem to be things that should be done *in Discord*.

### Checklist:
- [ ] Admin
  - [x] Announce
    - [X] Begin announcement
    - [x] Ignore announcements
    - [x] Set channel for announcements
  - [X] Serverlock
  - [x] Role commands
- [ ] Alias
  - [x] Global Aliases
    - [x] Make global aliases
    - [x] Delete global aliases
    - [x] List global aliases
  - [x] Guild-specific aliases
- [ ] Audio (TBD, look at Notes section)
- [ ] Bank
  - [ ] Toggle global bank
  - [ ] Change bank name
  - [ ] Change bank credits' name
- [ ] Cleanup (TBD, look at Notes section)
- [ ] Custom Commands
  - [ ] Create custom command
  - [ ] Edit custom command
  - [ ] Delete custom command
  - [ ] List custom commands
- [ ] Downloader (TBD, look at Notes section)
- [ ] Bank
  - [ ] Bank reset
  - [ ] Economy settings
    - [ ] Set minimum slot machine bid
    - [ ] Set maximum slot machine bid
    - [ ] Set cooldown for slot machine
    - [ ] Set cooldown for payday
    - [ ] Set amount for payday
    - [ ] Set amount earned based on roles for payday
    - [ ] Set initial balance for new bank accounts
- [ ] Filter
  - [ ] Autoban
  - [ ] Add filtered words to channel/guild
  - [ ] Remove filtered words from channel/guild
  - [ ] Toggle name/nickname filtering
- [ ] Image
  - [ ] Set imgur credentials
- [ ] Mod/Move To Core
  - [ ] Ignore a channel/guild
  - [ ] Unignore a channel/guild
- [ ] Mod
  - [ ] Disable/enable role hierarchy
  - [ ] Ban mention spam
  - [ ] Delete repeated messages
  - [ ] Delete delay
  - [ ] Reinvite unbanned users
- [ ] Modlog
  - [ ] Set a certain channel to be the modlog
  - [ ] Change settings for cases
  - [ ] Reset modlog cases for a guild
  - [ ] Set a reason for a case
- [ ] Permissions
  - [ ] Add/remove global rules
  - [ ] Add/remove guild rules
  - [ ] Set/clear default global rules
  - [ ] Set/clear default guild rules
- [ ] Reports
  - [ ] Set channel for reports
  - [ ] Enable/disable reports for a guild
- [ ] Streams
  - [ ] Manage channels + adding alerts for channels, and other settings
  - [ ] API key settings
- [ ] Trivia
  - [ ] Max score needed to win
  - [ ] Time limit
  - [ ] Stop after
  - [ ] Allow/disallow lists to override settings
  - [ ] Whether bot gets points
  - [ ] Whether or not the answer gets revealed
  - [ ] Payout multiplier
- [ ] Warnings
  - [ ] Allow/disallow custom reasons
  - [ ] Create/delete reasons
  - [ ] Get all registered reasons

This is only a rough outline of what I *plan* to create with RPC, but obviously this will take a very long time, judged by the length of the list.  I'm interested to here what you guys think about this, and which ones to actually create and which ones not too.

**Comment by yamikaitou** [comment]
Curious if it would be possible to maybe add a way to change the IP and Port that RPC binds to? I would love to be able to bind it to my DO Private IP so I can talk to it from my other DO hosts.

**Comment by NeuroAssassin** [comment]
Closing for the time being due to #3862
